controller/workitem_blackbox_test.go

```diff
@@ -2539,3 +2521,50 @@ func (s *workItemChildSuite) TestWorkItemListFilterByNoParents() {
 	})
 
 }
+
+// TestCreateMultipleWIs create multiple work items in parallel in different spaces to be sure that the implementation supports
+// concurrent writes (micro stress test)
+func (s *WorkItemSuite) TestCreateMultipleWIs() {
+	// given
+	spaceCount := 4
+	workitemsPerSpace := 4
+	results := make(chan bool)
+	// when
+	for i := 0; i < spaceCount; i++ {
+		// Create space
+		spacePayload := CreateSpacePayload("test-space"+uuid.NewV4().String(), "description")
+		_, testSpace := test.CreateSpaceCreated(s.T(), s.svc.Context, s.svc, s.spaceCtrl, spacePayload)
+		log.Info(nil, nil, "Created space with ID %v", *testSpace.Data.ID)
+		for j := 0; j < workitemsPerSpace; j++ {
+			go func() {
+				payload := minimumRequiredCreateWithTypeAndSpace(workitem.SystemBug, *testSpace.Data.ID)
+				payload.Data.Attributes[workitem.SystemTitle] = "Test WI" + uuid.NewV4().String()
+				payload.Data.Attributes[workitem.SystemState] = workitem.SystemStateNew
+
+				log.Info(nil, nil, "Creating a work item in space %v", *payload.Data.Relationships.Space.Data.ID)
+				// each go routine needs its own goa service.
+				svc := testsupport.ServiceAsUser("TestUpdateWI-Service", almtoken.NewManagerWithPrivateKey(s.priKey), s.testIdentity)
+				// when using a work item controller (ie, with its own Tx)
+				res, createdWorkitem := test.CreateWorkitemCreated(s.T(), svc.Context, svc, s.workitemCtrl, *payload.Data.Relationships.Space.Data.ID, &payload)
+				// then
+				if createdWorkitem == nil {
+					// problem occurred
+					log.Info(nil, nil, "Work item creation failed", res)
+					results <- false
+				} else {
+					// work item was created
+					log.Info(nil, nil, "Work item creation succeeded")
+					results <- true
+				}
+			}()
+		}
+	}
+	// then expect (spaceCount * workitemsPerSpace) responses
+	allGood := true
+	for r := 0; r < (spaceCount * workitemsPerSpace); r++ {
+		result := <-results
+		log.Info(nil, nil, fmt.Sprintf("Work item creation result #%d/%d: %t", (r+1), (spaceCount*workitemsPerSpace), result))
+		allGood = allGood && result
```

Comment:
I wasn't sure about the behavior of `assert.XYZ()` functions within a goroutine, to be honest. Let me verify that and apply your suggestion.